### PR TITLE
Document Aggregation Fuzzer

### PR DESCRIPTION
**Pull request by mbasmanova**
Also, move Fuzzer documentation under Developer Guide / Testing Tools (from Developer Guide / Debugging Tools).

### CODE REVIEW

File: velox/docs/develop/testing/fuzzer.rst [hunk 1]
```diff
@@ -1,64 +1,106 @@
...
+aggregations (one or more grouping keys), distinct aggregations(no aggregates),
+aggregations with and without masks.
+
+The results of aggregations using functions supported by DuckDB are compared
+with DuckDB results.
+
+For each aggregation, Fuzzer generates multiple logically equivalent plans and
+verifies that results match. These plans are:
+
+- Single aggregation (raw input, final result).
```
Comment (xiaoxmeng) [inline velox/docs/develop/testing/fuzzer.rst:34]: Do we plan to cover spilling next? Thanks!

File: velox/docs/develop/testing/fuzzer.rst [hunk 2]
```diff
@@ -1,64 +1,106 @@
...
+- Partial -> Final aggregation.
+- Partial -> Intermediate -> Final aggregation.
+- Partial -> LocalExchange(grouping keys) -> Final aggregation.
+- All the above using flattened input vectors.
+
+When testing aggregate functions whose results depend on the order of inputs
+(e.g. map_agg, map_union, arbitrary, etc.), the Fuzzer verifies that all plans
+succeed or fail with compatible exceptions. When plans succeed, the Fuzzer
```
Comment (xiaoxmeng) [inline velox/docs/develop/testing/fuzzer.rst:42]:
```
s/compatible exceptions/compatible user exceptions/?
```

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova  Nice document. Thanks!

File: velox/docs/develop/testing/fuzzer.rst (hunk 1 above)
Comment (mbasmanova) [reply velox/docs/develop/testing/fuzzer.rst:34]: That would be nice: #3204

**Pull request closed without merging**